src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -6,6 +6,38 @@ namespace System.Collections.Tests
 {
     public abstract partial class Dictionary_Generic_Tests<TKey, TValue> : IDictionary_Generic_Tests<TKey, TValue>
     {
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_TryAdd(int count)
+        {
+            var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+
+            TKey key = CreateTKey(seed: count);
+            TValue value = dictionary.ContainsKey(key) ? dictionary[key] : default(TValue);
+            int originalCount = dictionary.Count;
+            bool alreadyContainsKey = dictionary.ContainsKey(key);
```

Comment:
@stephentoub 

> What does it have to do with GetValueOrDefault?

`TValue value = dictionary.GetValueOrDefault(key)`

> Why did you ever need to call it twice?

I didn't think it mattered because this is test code. Are you concerned about code duplication in calling the method twice?